packages/lesswrong/lib/collections/electionVotes/schema.ts

```diff
@@ -0,0 +1,51 @@
+import { foreignKeyField } from "../../utils/schemaUtils";
+import { userOwns } from "../../vulcan-users";
+
+const schema: SchemaType<DbElectionVote> = {
+  /** The name of the election */
+  electionName: {
+    type: String,
+    optional: false,
+    nullable: false,
+    canRead: [userOwns, "sunshineRegiment", "admins"],
+    canCreate: ["members"],
+    canUpdate: [userOwns, "sunshineRegiment", "admins"],
+  },
+  /** The user voting */
+  userId: {
+    ...foreignKeyField({
+      idFieldName: "userId",
+      resolverName: "user",
+      collectionName: "Users",
+      type: "User",
+      nullable: false,
+    }),
+    canRead: [userOwns, "sunshineRegiment", "admins"],
+    canCreate: ["members"],
+    canUpdate: [userOwns, "sunshineRegiment", "admins"],
+  },
+  /**
+   * Object like {'cF8iwCmwFjbmCqYkQ': 1, 'cF8iwCmwFjbmCqYkQ': 2} representing the (unnormalised) weights
+   * given to each selected candidate. Weights can not be negative, but can be null (these will
+   * be ignored).
+   */
```

Comment:
Worth reading this, this is the actual vote object. When you select candidates in the select page, it adds them to here with the weight set to null, and then the weights will be filled in in the allocation page